__tests__/Unit/Components/Tasks/DependencyList.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import DependencyList from '@/components/taskDetails/taskDependency/DependencyList';
+import { taskDetailsHandler } from '../../../../__mocks__/handlers/task-details.handler';
+import { setupServer } from 'msw/node';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+
+const taskDependencyIds = ['6KhcLU3yr45dzjQIVm0J', 'taskid-2'];
+const mockNavigateToTask = jest.fn();
+jest.mock('next/router', () => ({
+    useRouter: () => ({
+        push: mockNavigateToTask,
+    }),
+}));
+describe('DependencyList', () => {
+    const server = setupServer(...taskDetailsHandler);
+    beforeAll(() => server.listen());
+    afterEach(() => server.resetHandlers());
+    afterAll(() => server.close());
+
+    it('should render loading state', () => {
+        render(
+            <Provider store={store()}>
+                <DependencyList taskDependencyIds={taskDependencyIds} />
+            </Provider>
+        );
+
+        expect(screen.getByText(/Loading/i)).toBeInTheDocument();
+    });
+
+    it('should render dependency list', async () => {
+        render(
+            <Provider store={store()}>
+                <DependencyList taskDependencyIds={taskDependencyIds} />
+            </Provider>
+        );
+
+        const task1Link = await screen.findByText('test 1 for drag and drop');
+        const errorMessage = await screen.findByText(
+            'Unable to fetch this task with ID taskid-2'
+        );
+        const dependencyItems = await screen.findAllByRole('listitem');
+        expect(dependencyItems).toHaveLength(2);
+
+        expect(task1Link).toBeInTheDocument();
+        expect(errorMessage).toBeInTheDocument();
+    });
+
+    it('should render no dependencies message', () => {
```

Comment:
You're testing for loading here. What about "no dependencies message"

Reply:
done